homeworks/nagynorbert/5_oop/car_fleet_mpr_classes.py

```diff
@@ -0,0 +1,83 @@
+# Homework 5. Nagy Norbert
+
+class OutOfFuelError(Exception):
+    """Fuel is not enough for the defined distance."""
+    pass
+
+class TooMuchFuelError(Exception):
+    """Fuel is too much."""
+    pass
+
+class CarDefineError(Exception):
+    """Number of parameters for defining a car is not met with specification."""
+    pass
+
+class UserDefineError(Exception):
+    """Number of how many cars should manipulate is incorrect."""
+    pass
+
+class Car:
+    def __init__(self,brand:str,model:str,year:int):
+        self.brand = brand
+        self.model = model
+        self.year = year
+        self.mileage = 0
+        self.fuel_level = 100
+    
+    def __str__(self):
+        return f"Brand:{self.brand} -- Model:{self.model} -- Year:{self.year} -- Mileage:{self.mileage} -- Fuel_level:{self.fuel_level}"
+
+    def drive(self,km:float):
+        fuel_consumption = 0.1 * km
+        if fuel_consumption > self.fuel_level:
+            raise OutOfFuelError("Fuel is not enough in car.")
+        self.mileage += km
+        print(f"Mileage now: {self.mileage}")
+        self.fuel_level -= fuel_consumption
+        return True
+
+    def refuel(self,fuel_amount:float):
+        fuel_can_add = self.fuel_level + fuel_amount
+        if fuel_can_add > 100:
+            raise TooMuchFuelError("The amount of refuel is too much.")
+        self.fuel_level += fuel_amount
+        return True
+    
+class Fleet:
+    cars_in_fleet = []
```

Comment:
Mivel most egy flottánk van, teljesen jó hogy ez egy osztály szintű változó, de ha több flottának kezelnéd az autóit, akkor ez egy instance variable kéne legyen.

Reply:
Jogos, áthelyeztem.